src/MLJInterface.jl

```diff
@@ -264,14 +267,19 @@ function update(mlj_model::MLJInterface.MODELS, verbosity::Int, fitresult, cache
     old_lgbm_model.num_iterations = num_iterations + additional_iterations
 
     # eh this is ugly, possibly prompts a need for some refactoring
-    train_metrics = LightGBM.train!(old_lgbm_model, additional_iterations, String[], verbosity, LightGBM.Dates.now())
+    train_results = LightGBM.train!(old_lgbm_model, additional_iterations, String[], verbosity, LightGBM.Dates.now(), truncate_booster=old_mlj_model.truncate_booster)
     fitresult = (old_lgbm_model, old_classes, deepcopy(mlj_model))
 
     final_num_iter = LightGBM.get_iter_number(old_lgbm_model)
     iteration_history = deepcopy(cache.num_boostings_done)
     push!(iteration_history, final_num_iter - num_iterations)
-    report = user_fitreport(old_lgbm_model, cache.training_metrics, train_metrics)
-    newcache = (num_boostings_done=iteration_history, training_metrics=report.training_metrics)
+
+    report = user_fitreport(old_lgbm_model, cache.training_metrics, train_results)
+    newcache = (
+        num_boostings_done=iteration_history, 
+        training_metrics=report.training_metrics,
+        best_iter = report.best_iter,
```

Comment:
Why does this need to be in the cache?

Reply:
you're right it doesn't. Will remove this. thanks!